airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -182,7 +174,59 @@ def test_oauth_flow_parameters(self, actual_connector_spec: ConnectorSpecificati
         diff = params - schema_path
         assert diff == set(), f"Specified oauth fields are missed from spec schema: {diff}"
 
-    def test_additional_properties_is_true(self, actual_connector_spec):
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_new_required_field_declaration(self, spec_diff):
+        added_required_fields = [
+            addition for addition in spec_diff.get("dictionary_item_added", []) if addition.path(output_format="list")[-1] == "required"
+        ]
+        assert len(added_required_fields) == 0, f"The current spec has a new required field: {spec_diff.pretty()}"
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_new_required_property(self, spec_diff):
+        added_required_properties = [
+            addition for addition in spec_diff.get("iterable_item_added", []) if addition.up.path(output_format="list")[-1] == "required"
+        ]
+        assert len(added_required_properties) == 0, f"The current spec has a new required property: {spec_diff.pretty()}"
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_type_field_changed_from_list_to_string(self, spec_diff):
+        type_changes = [
+            type_change for type_change in spec_diff.get("type_changes", []) if type_change.path(output_format="list")[-1] == "type"
+        ]
+        for type_change in type_changes:
+            if isinstance(type_change.t1, list) and isinstance(type_change.t2, str):
+                raise AssertionError(f"The current spec narrowed a field type: {spec_diff.pretty()}")
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_type_field_has_narrowed(self, spec_diff):
```

Comment:
Is there a missing test here for changing the type entirely? (e.g. `string` -> `object` OR `["string"]` -> `["object"]`). The array case might be handled by this narrowing test, but I think we may be missing one for strings.

Reply:
I tried to be as exhaustive as I could by adding a new test: `test_field_type_changed`